src/public/publicAPIs.ts

```diff
@@ -46,12 +46,13 @@ export function initialize(callback?: () => void, validMessageOrigins?: string[]
         ? GlobalVars.currentWindow.parent
         : GlobalVars.currentWindow.opener;
 
+    // Listen to messages from the parent or child frame.
+    // Frameless windows will only receive this event from child frames.
+    GlobalVars.currentWindow.addEventListener('message', messageListener, false);
```

Comment:
@Kanac, just to be safe would it be possible to put this call behind an explicit check. Either:
1. We can register it anytime `validMessageOrigins` is specified:
    ```
    if (validMessageOrigins) {
      // Register message handler
    }
    ```
2. Or we can add a new parameter to the `initialize` API - something like `enableChildFrameCommunication?: boolean`.

What do you think? Would either of these approaches work better for your use case?

Reply:
For #1, I thought the message wouldn't get pass the check here https://github.com/OfficeDev/microsoft-teams-library-js/blob/f3d43bc7f1953e010e340e39d21133bf34479bbc/src/internal/internalAPIs.ts#L193 if `validMessageOrigins` isn't defined. But then I noticed we have a bunch of default `validMessageOrigins` here https://github.com/OfficeDev/microsoft-teams-library-js/blob/69962f9344710a7a1f5a7b80b3f6f48712854f08/src/internal/constants.ts#L12 that would still be allowed to communicate with the frame. I'm guessing those won't cause a problem for the frameless case, but I guess we can add this check here.

for #2, I think that would be a breaking change for any existing users that already have child frame communication that would need to now remember to pass in that parameter.